src/QtImageReader.cpp

```diff
@@ -269,22 +297,28 @@ bool QtImageReader::load_svg_path(QString) {
         image = std::make_shared<QImage>();
         loaded = image->load(path);
 
-        if (loaded && (image->width() < current_max_size.width() || image->height() < current_max_size.height())) {
-            // Load SVG into larger/project size (so image is not blurry)
-            QSize svg_size = image->size().scaled(current_max_size.width(), current_max_size.height(), Qt::KeepAspectRatio);
-            if (QCoreApplication::instance()) {
-                // Requires QApplication to be running (for QPixmap support)
-                // Re-rasterize SVG image to max size
-                image = std::make_shared<QImage>(QIcon(path).pixmap(svg_size).toImage());
-            } else {
-                // Scale image without re-rasterizing it (due to lack of QApplication)
-                image = std::make_shared<QImage>(image->scaled(
-                        svg_size.width(), svg_size.height(), Qt::KeepAspectRatio, Qt::SmoothTransformation));
+        if (loaded) {
+            // Set default SVG size
+            default_size.setWidth(image->width());
+            default_size.setHeight(image->height());
+
+            if (image->width() < current_max_size.width() || image->height() < current_max_size.height()) {
+                // Load SVG into larger/project size (so image is not blurry)
+                QSize svg_size = image->size().scaled(current_max_size.width(), current_max_size.height(), Qt::KeepAspectRatio);
```

Comment:
If I understand this, It's rendering the SVG at the maximum resolution, so we can scale down to the preview size... Let me know if I'm way off.

Reply:
Yes, all of this logic is based on the concept that our preview window might be "smaller" than the project, and thus, we can cheat and use scaled down assets (videos, images, SVGs), so things go smoother during real-time preview.